pixeltable/io/pandas.py

```diff
@@ -0,0 +1,144 @@
+from typing import Optional, Any, Iterable
+
+import numpy as np
+
+import pandas as pd
+import pixeltable as pxt
+import pixeltable.exceptions as excs
+import pixeltable.type_system as ts
+
+
+def import_pandas(
+        tbl_name: str,
+        df: pd.DataFrame,
+        *,
+        schema: Optional[dict[str, pxt.ColumnType]] = None
+) -> pxt.catalog.InsertableTable:
+    if schema is None:
+        # Infer schema
+        schema = _df_to_pxt_schema(df)
+    else:
+        # Validate the specified schema
+        if len(schema) != len(df.columns):
```

Comment:
I don't think we should insist on that. In pd.read_csv(), the dtype parameter lets you override specific columns. It's pretty annoying to have to specify the full schema of a 20-column df when only 3 columns have dtype object and need to be overridden.